File: algorithms/ruby/sorting/quick_sort.rb
def quick_sort(arr)
  return arr if arr.length <= 1

  pivot = arr.delete_at(rand(arr.length))
  left, right = arr.partition { |x| x < pivot }

  return *quick_sort(left), pivot, *quick_sort(right)
end

if __FILE__ == $0
  arr = [10, 7, 8, 9, 1, 5]
  puts "Original array: #{arr.inspect}"
  sorted_arr = quick_sort(arr)
  puts "Sorted array: #{sorted_arr.inspect}"
end

File: algorithms/julia/sorting/quick_sort.jl
function quick_sort(arr)
    if length(arr) <= 1
        return arr
    end
    
    pivot = arr[end]
    left = filter(x -> x < pivot, arr[1:end-1])
    right = filter(x -> x >= pivot, arr[1:end-1])
    
    return vcat(quick_sort(left), pivot, quick_sort(right))
end

arr = [10, 7, 8, 9, 1, 5]
println("Original array: ", arr)
sorted_arr = quick_sort(arr)
println("Sorted array: ", sorted_arr)
